### PR TITLE
feat(Common/IPLocation): Replace ip2nation by ip2location

### LINKED ISSUE #5631: Replace ip2nation to ip2location [$10]
## CURRENT BEHAVIOUR:

ip2nation is outdated, it was last updated over a year ago whilst ip2location was last updated 30-Apr-2021

Support IPV4 CSV only

https://github.com/TrinityCore/TrinityCore/commit/2fe4ab94c52ad2fa90189ecd9c053da49f06561e

## AC HASH/COMMIT:

https://github.com/azerothcore/azerothcore-wotlk/commit/f8b817a8fef40cc4f66ef534f1a15d7c7ab0808f

<bountysource-plugin>

---
There is a **[$10 open bounty](https://www.bountysource.com/issues/98272610-replace-ip2nation-to-ip2location?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Download
- https://download.ip2location.com/lite/ `IP2LOCATION-LITE-DB1.CSV`
- Full URL https://download.ip2location.com/lite/IP2LOCATION-LITE-DB1.CSV.ZIP

## Changes Proposed:
- https://github.com/TrinityCore/TrinityCore/commit/2fe4ab94c52ad2fa90189ecd9c053da49f06561e

## Issues Addressed:
- Close #5631

## Tests Performed:
- In-game

## How to Test the Changes:
- Download and set in config files `IPLocationFile`
- Set debug or trace level logger `root`
- Enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to country
```
- `.acc l c on`
- Try to re-enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is locked to country: '-' Player country is '-'
```
- `.acc l c off`
- Try to re-enter in-game
- Check `authserver` console, like this:
```
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to ip
[AuthChallenge] Account 'WINFIDONARLEYAN' is not locked to country
```

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Dont forget to remove from base files too

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/5653#issuecomment-832900455

**Comment by Winfidonarleyan** [comment]
> [#5653 (comment)](https://github.com/azerothcore/azerothcore-wotlk/pull/5653#issuecomment-832900455)

I don't understand what you want :D

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/base/db_auth/ip2nation.sql
https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/base/db_auth/ip2nationCountries.sql

**Comment by Winfidonarleyan** [comment]
> https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/base/db_auth/ip2nation.sql
> https://github.com/azerothcore/azerothcore-wotlk/blob/master/data/sql/base/db_auth/ip2nationCountries.sql
> delete

Ahah, this, yes yes, of course

**Comment by Winfidonarleyan** [comment]
Done

File: src/common/IPLocation/IPLocation.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3
+ * Copyright (C) 2021+ WarheadCore <https://github.com/WarheadCore>
```
Comment (Kitzunu) [inline src/common/IPLocation/IPLocation.cpp:3]: Should also include TC?

File: src/common/IPLocation/IPLocation.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2016+ AzerothCore <www.azerothcore.org>, released under GNU AGPL v3 license: https://github.com/azerothcore/azerothcore-wotlk/blob/master/LICENSE-AGPL3
+ * Copyright (C) 2021+ WarheadCore <https://github.com/WarheadCore>
```
Comment (Kitzunu) [inline src/common/IPLocation/IPLocation.h:3]: ssame

File: src/common/IPLocation/IPLocation.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/common/IPLocation/IPLocation.cpp:3]: i use https://github.com/WarheadCore/Warhead, fork TC

**Comment by Winfidonarleyan** [comment]
Now no work `authserver`

File: src/common/IPLocation/IPLocation.h (hunk 2 above)
Comment (Winfidonarleyan) [reply src/common/IPLocation/IPLocation.h:3]: TC fork

**Comment by Winfidonarleyan** [comment]
Need retest this

**Review by Kitzunu — APPROVED** [review]
Works and looks good =)

**Pull request merged**